### PR TITLE
fix: prevent crash when running tests on config with empty rules list

### PR DESCRIPTION
**Pull request by jtimmons**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

## Description
Fixes a small edge cases where `semgrep --test` fails when running against a yaml file with no rules in it - this is because the `config_contains_fix_key` check assumes that there's at least one rule in the rules list. Found this after our team disabled a subset of our tests.

example rule file to test against:
```yaml
rules: []
```

### CODE REVIEW

**Comment by brendongo** [comment]
I don't think this would fix the bug you described. `rules: []` will still pass that condition. You probably want to add a check that the list is non empty before the first element access https://github.com/returntocorp/semgrep/pull/6954/files#diff-26dadf2b538a85df720f7872378ff02d0755ed5f11b3d4e5ea2ac64a0486a605R418

**Comment by jtimmons** [comment]
> I don't think this would fix the bug you described. `rules: []` will still pass that condition. You probably want to add a check that the list is non empty before the first element access https://github.com/returntocorp/semgrep/pull/6954/files#diff-26dadf2b538a85df720f7872378ff02d0755ed5f11b3d4e5ea2ac64a0486a605R418

I confirmed that it fixes our issue locally! The existing check is only for the existence of the `"rules"` key but since `.get()` returns the actual value (if it exists) this will check its truthiness. An empty array is falsey so it hits the `else` condition

**Comment by brendongo** [comment]
Ah yes of course. empty list is non truthy. 😅

**Review by brendongo — APPROVED** [review]

**Comment by brendongo** [comment]
Mind adding a changelog entry to the PR and we can get this merged

**Comment by jtimmons** [comment]
> Mind adding a changelog entry to the PR and we can get this merged

done!

File: changelog.d/gh-6954.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+CLI: No longer fails when running tests against a config file with no rules in it
```
Comment (brendongo) [inline changelog.d/gh-6954.fixed]:
```suggestion
CLI: No longer fails when running tests against a config file with no rules in it

```

**Comment by brendongo** [comment]
Thanks @jtimmons
